library/src/webapp-filtered/editor/ckeditor.launch.js

```diff
@@ -30,6 +30,21 @@ sakai.editor.editors.ckeditor = sakai.editor.editors.ckeditor || {} ;
 var basePath = "/library/editor/ckextraplugins/";
 var webJars = "/library/webjars/"
 
+// Update properties in one object from another: https://stackoverflow.com/a/12534361/3708872
+// I believe this is available as lodash.merge but don't see that available here yet and this looked like the simplest version of that
+function objectMerge(obj/*, ...*/) {
```

Comment:
does it make sense to use the es6 spread operator here? if so there are a few other locations that do this and could also be updated.

Reply:
The problem is that this needs to be a deep extend/merge.

I was [reading on this](https://stackoverflow.com/questions/171251/how-can-i-merge-properties-of-two-javascript-objects-dynamically) that the only way to do this is with a custom code solution or a larger library like jQuery or lodash currently

The problem with jQuery is it's not always guaranteed to be loaded before CKEditor. There have been some plugins in the past that have loaded jQuery with `CKEDITOR.scriptLoader.load` but I felt like using less jQuery nowadays is preferred. I also only see lodash used somewhat in web components. 

spread and Object.assign both seem to only make shallow copies and if there is an object will overwrite the entire object. My idea is I'd only pass through the change that would be made rather than the entire new object. (I guess this could be an option)

I was experimenting a little too much here: https://jsfiddle.net/jonespm/ste45hko/4/

We could load jQuery or try some other library I guess, but this code seemed to be working and pretty simple.

Reply:
thanks for the explanantion